modeling.md

```diff
@@ -171,18 +175,28 @@ keys are “foreign” keys from the referenced entities. Instead of using
 Create, new association relationships are made by using Update, which
 allows the client to provide the resource identifier.
 
+Asynchronous association resources are declared by creating a class that extends AssociationResourceAsyncTemplate or AssociationResourceTaskTemplate.
+
 For example:
 
 ```java  
 @RestLiAssociation(name="myRelations", assocKeys={`Key(name=“key1”, type=long.class), @Key(name=“key2”, type=long.class)}, …)  
-public class MyRelationResource extends AssociationResourceTemplate<Relation> { … }  
+public class MyRelationResource extends AssociationResourceTaskTemplate<Relation> { … }  
 ```
 
 Defines a child resource with a URI of the form:
 
 ```  
 /myRelations/key1={longKey1}\&key2={longKey2}  
 ```
+#### Asynchronous resources
+Asynchronous Rest.li resources should be used when the downstream implementations are non-blocking. There are asynchronous resource templates for Collection, Simple, Association, and ComplexKey resources.
```

Comment:
What do you mean "when the downstream implementations are non-blocking"? Like if a service downstream from your resource uses an async template? May need a little more clarification

Reply:
I added a little more clarification. The page I linked at the bottom of these section goes more into detail about asynchronous, as well.